proposals/0253-callable.md

```diff
@@ -669,6 +676,24 @@ Adder.self(base: 3) // okay
 
 But since this would be an additive feature on top of this proposal and that `subscript` cannot be `static` yet, we'd like to defer this feature to future discussions.
 
+### Direct references to `call` members
+
+Direct references to `call` members are intentionally not supported. An earlier
+version of the proposal included support for direct references to `call` members
+via `foo.call` (like referring to an instance method with the name "call").
+
+However, the direction for "callable values" is not to support direct `call`
+member references, but to support conversions of callable values to
+function-typed values. Supporting explicit conversion via `as` seems relatively
+non-controversial in forum discussions. Implicit conversion was a highly
+requested feature, but it likely has type-system-wide impact and requires more
+exploration.
+
+`call` members should be thought of and represented as "instance methods with a
+special-case name", not "instance methods with the name 'call'". For now,
+without support for conversion to function-typed values, create thunks like `{

```

Comment:
What are trunks?

Reply:
The terminology is actually ["thunks"](https://en.wikipedia.org/wiki/Thunk), which are like small helper functions.

Actually, a better name is just "closures". Swift user-facing documentation doesn't use the term "thunk".